src/library/Uno.Material/Controls/Chips/ChipGroup.cs

```diff
@@ -0,0 +1,332 @@
+					container.ThumbnailTemplate = ThumbnailTemplate;
+				}
+			}
+		}
+
+		private void ApplyCanRemoveProperty()
+		{
+			foreach (var container in GetItemContainers())
+			{
+				container.CanRemove = CanRemove;
+			}
+		}
+
+		private void OnSelectionModeChanged(DependencyPropertyChangedEventArgs e)
+		{
+			EnforceSelectionMode();
+		}
+
+		private void OnItemIsCheckedChanged(object sender, RoutedEventArgs e)
+		{
+			if (_isSynchronizingSelection) return;
+			if (sender is Chip container)
+			{
+				UpdateSelection(new[] { container });
+			}
+		}
+
+		protected override void OnItemsChanged(object e)
+		{
+			base.OnItemsChanged(e);
+
+			EnforceSelectionMode();
+		}
+
+		protected void OnItemsSourceChanged()
+		{
+			EnforceSelectionMode();
+		}
+
+		private void OnSelectedItemChanged(DependencyPropertyChangedEventArgs e)
+		{
+			if (_isSynchronizingSelection || _isUpdatingSelection) return;
+
+			if (SelectionMode == ChipSelectionMode.Single && FindContainer(SelectedItem) is Chip container)
+			{
+				container.SetIsCheckedSilently(true);
+				UpdateSelection(new[] { container });
+			}
+			else
+			{
+				UpdateSelection(null);
+			}
+		}
+
+		private void OnSelectedItemsChanged(DependencyPropertyChangedEventArgs e)
+		{
+			if (_isSynchronizingSelection || _isUpdatingSelection) return;
+
+			if (SelectionMode == ChipSelectionMode.Multiple)
+			{
+				var selectedContainers = SelectedItems != null
+					? GetItemContainers().Where(x => SelectedItems.Contains(x)).ToArray()
+					: null;
+				foreach (var container in selectedContainers ?? Enumerable.Empty<Chip>())
+				{
+					container.SetIsCheckedSilently(true);
+				}
+
+				UpdateSelection(selectedContainers, forceClearOthersSelection: true);
+			}
+			else
+			{
+				UpdateSelection(null, forceClearOthersSelection: true);
+			}
+
+		}
+
+		protected override bool IsItemItsOwnContainerOverride(object item) => item is Chip;
+
+		protected override DependencyObject GetContainerForItemOverride() => new Chip();
+
+		protected override void PrepareContainerForItemOverride(DependencyObject element, object item)
+		{
+			base.PrepareContainerForItemOverride(element, item);
+
+			if (element is Chip container)
+			{
+				if (SelectionMemberPath != null)
+				{
+					container.SetBinding(Chip.IsCheckedProperty, new Binding { Path = new PropertyPath(SelectionMemberPath), Mode = BindingMode.TwoWay });
+				}
+
+				if (ThumbnailTemplate != null)
+				{
+					container.Thumbnail = container.Content;
+					container.ThumbnailTemplate = ThumbnailTemplate;
+				}
+
+				container.CanRemove = CanRemove;
+
+				container.IsCheckedChanged += OnItemIsCheckedChanged;
+				container.Click += OnItemClick;
+				container.Checked += OnItemChecked;
+				container.Unchecked += OnItemUnchecked;
+			}
+		}
+
+		protected override void ClearContainerForItemOverride(DependencyObject element, object item)
+		{
+			base.ClearContainerForItemOverride(element, item);
+
+			if (element is Chip container)
```

Comment:
How does this statement work? When does it resolve to true?

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/is#type-pattern